src/app/+detail/index.ts

```diff
@@ -20,8 +20,9 @@ export const routes = [
     BrowserModule,
     FormsModule,
     RouterModule.forChild(routes),
+    MdModule.forRoot()
```

Comment:
You don't need to use .forRoot() on MdModule since md providers are already loaded by app